aten/src/ATen/Parallel.h

```diff
@@ -1,6 +1,8 @@
 #pragma once
 #include <ATen/ATen.h>
 
+#include <future>
```

Comment:
I don't know how heavy of an include this is. If Parallel.h is supposed to be used widely it might make sense to move this somewhere else.

Reply:
Maybe create a Future.h?

Reply:
discussed offline, will follow up with a more comprehensive reorg of parallel code files